mathjax2/legacy/jax/input/TeX/jax.js

```diff
@@ -1070,25 +451,33 @@
    *   The TeX Parser
    */
 
+  var NewParser = new TeXParser();
   var PARSE = MathJax.Object.Subclass({
+    remap:   MapHandler.getInstance().getMap('remap'),
     Init: function (string,env) {
       this.string = string; this.i = 0; this.macroCount = 0;
       var ENV; if (env) {ENV = {}; for (var id in env) {if (env.hasOwnProperty(id)) {ENV[id] = env[id]}}}
       this.stack = TEX.Stack(ENV,!!env);
+      NewParser.setup(this);
```

Comment:
Indeed this will be dealt with more efficiently in the alpha version. New instances of TexParser will only contain a reference to the symbol mappings. So you can reorder them if you want for a recursive call, but they do not have copies of the actual objects.